mso/datasource_mso_schema_template_service_graph.go

```diff
@@ -35,40 +33,35 @@ func dataSourceMSOSchemaTemplateServiceGraph() *schema.Resource {
 				Required:     true,
 				ValidateFunc: validation.StringLenBetween(1, 1000),
 			},
-			"site_nodes": &schema.Schema{
-				Type:     schema.TypeList,
-				Computed: true,
+			"service_node_type": &schema.Schema{
+				Type:          schema.TypeString,
+				Computed:      true,
+				ConflictsWith: []string{"service_node"},
+				Deprecated:    "Use service_node to configure service nodes.",
+			},
+			"service_node": &schema.Schema{
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: "Configure service nodes for the service graph.",
 				Elem: &schema.Resource{
 					Schema: map[string]*schema.Schema{
-						"node_index": &schema.Schema{
-							Type:     schema.TypeInt,
-							Computed: true,
-						},
-						"service_node_type": &schema.Schema{
-							Type:     schema.TypeString,
-							Computed: true,
-						},
-						"site_id": &schema.Schema{
-							Type:     schema.TypeString,
-							Computed: true,
-						},
-						"tenant_name": &schema.Schema{
-							Type:     schema.TypeString,
-							Computed: true,
-						},
-						"node_name": &schema.Schema{
+						"type": &schema.Schema{
 							Type:     schema.TypeString,
 							Computed: true,
 						},
 					},
 				},
 			},
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
 		}),
 	}
 }
 
-func dataSourceMSOSchemaTemplateServiceGrapRead(d *schema.ResourceData, m interface{}) error {
-	log.Printf("[DEBUG] %s: Beginning Read", d.Id())
+func dataSourceMSOSchemaTemplateServiceGraphRead(d *schema.ResourceData, m interface{}) error {
+	log.Printf("[DEBUG] %s: Beginning datasource Read", d.Id())
```

Comment:
Don't think id is known here
```suggestion
	log.Printf("[DEBUG] Beginning datasource Read")
```